frontend/build.go

```diff
@@ -59,6 +60,38 @@ func makeRequestHandler(target string) dockerui.RequestHandler {
 	return h
 }
 
+func getOS(platform ocispecs.Platform) string {
+	return platform.OS
+}
+
+func getArch(platform ocispecs.Platform) string {
+	return platform.Architecture
+}
+
+func getVariant(platform ocispecs.Platform) string {
+	return platform.Variant
+}
+
+func getPlatformFormat(platform ocispecs.Platform) string {
+	return platforms.Format(platform)
+}
+
+var passthroughGetters = map[string]func(ocispecs.Platform) string{
+	"OS":       getOS,
+	"ARCH":     getArch,
+	"VARIANT":  getVariant,
+	"PLATFORM": getPlatformFormat,
+}
+
+func fillPlatformArgs(prefix string, args map[string]string, platform ocispecs.Platform) map[string]string {
+	args = dalec.DuplicateMap(args)
```

Comment:
How come we are duplicating and returning a new map?

Reply:
This should probably be moved. When this function is used, it is initially passed `bc.BuildArgs` from the docker client. I didn't want to mutate the underlying `BuildArgs` map as that seems like confusing behavior, but I could instead do the copy before ever calling this function, leading to only one copy instead of a copy on **every** call.